### PR TITLE
More Generic Topping Logic

### PR DESCRIPTION
**Pull request by Stormsurfer36**
Toppings now contain up to 10 toppings. This is limited on the back end by Pizza.LIMIT. User sent alert for when they go over limit.
Radio buttons replaced javascript mess for topping position selecting.
Styling fixes(remove truncate of voted pizza titles). Sauce now under
crust type.
Fixed mobile view to not be janky.

### CODE REVIEW

**Comment by doeiqts** [comment]
Is that 10 topping limit a binary joke?

File: src/main/java/com/doeiqts/pizzavoter/domain/Pizza.java [hunk 1]
```diff
@@ -23,20 +29,29 @@ public Pizza(Size size, Crust crust, Sauce sauce) {
         this.size = size;
         this.sauce = sauce;
         toppings.put(Topping.CHEESE, Position.ALL);
+        //it's effectively 0 but cheese is on everything...not sure how I feel about this...
+        toppingCount = 1.0;
     }
 
     public Position addTopping(Topping topping, Position position) {
-        return this.toppings.put(topping,position);
+        double old = 0.0; //if duplicate topping positions are sent need to subtract existing topping value)
```
Comment (doeiqts) [inline src/main/java/com/doeiqts/pizzavoter/domain/Pizza.java]: I think we can clean up this logic by using the return value we get from Map.put() and checking the toppingCount against the LIMIT after we put the new value and just revert it if needed. IDK, not going to hold up the pull request, just commenting for thought later.

File: src/main/java/com/doeiqts/pizzavoter/servlet/PizzaVoterServlet.java [hunk 2]
```diff
@@ -138,17 +137,22 @@ private Pizza serializePizza(HttpServletRequest request, String pizzaNumber) {
...
             String[] toppings = request.getParameterValues("toppings" + pizzaNumber);
-            String[] positions = request.getParameterValues("position" + pizzaNumber);
             for (int i = 0; i < toppings.length; i++) {
-                if (toppings[i] != "") {
-                    Position newPosition = Position.valueOf(positions[i]);
-                    Position old = pizza.addTopping(Topping.valueOf(toppings[i]),newPosition);
-                    if(Position.ALL.equals(old) || (Position.RIGHT.equals(old) && newPosition.equals(Position.LEFT)) ||
-                        (Position.LEFT.equals(old) && newPosition.equals(Position.RIGHT))) {
-                        pizza.addTopping(Topping.valueOf(toppings[i]), Position.ALL);
+                if (toppings[i] != "" && pizza.getToppingCount() < Pizza.getToppingLimit()) {
```
Comment (doeiqts) [inline src/main/java/com/doeiqts/pizzavoter/servlet/PizzaVoterServlet.java]: Do we need to check the limits here if the domain object already handles that logic?

File: src/main/java/com/doeiqts/pizzavoter/servlet/PizzaVoterServlet.java [hunk 3]
```diff
@@ -138,17 +137,22 @@ private Pizza serializePizza(HttpServletRequest request, String pizzaNumber) {
...
+                    String[] positions = request.getParameterValues("position" + pizzaNumber + "-"+(i+1));
+                    if(null != positions) {
+                        Position newPosition = Position.valueOf(positions[0]);
+                        Position old = pizza.addTopping(Topping.valueOf(toppings[i]), newPosition);
+                        if (Position.ALL.equals(old) || (Position.RIGHT.equals(old) && newPosition.equals(Position.LEFT)) ||
```
Comment (doeiqts) [inline src/main/java/com/doeiqts/pizzavoter/servlet/PizzaVoterServlet.java]: I feel like this logic can be moved to the domain object as well.

File: src/main/webapp/pizzavoter.jsp [hunk 4]
```diff
@@ -52,10 +51,16 @@
...
+        <div class="alert alert-info alert-dismissible" role="alert">
+            <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+            Topping Limit of <strong>${limit}</strong> was exceeded. Excess toppings have been removed from your pizzas.
+        </div>
+        </c:if>
         <form role="form" name="vote" method="post" action="/">
             <div class="row">
                 <c:forEach begin="0" end="3" varStatus="pizza">
-                    <div class=" col-lg-3 col-md-3 col-sm-6 col-xs-6">
+                    <div class=" col-lg-3 col-md-3 col-sm-6 col-xs-8">
```
Comment (doeiqts) [inline src/main/webapp/pizzavoter.jsp]: col-xs-8 might as well be col-xs-12 since both push it to only 1 per row.

Comment (Stormsurfer36) [inline src/main/webapp/pizzavoter.jsp]: It looks shitty at full width. Like stretching an image in paint/photoshop.

**Pull request merged**